### PR TITLE
Change normalization

### PR DESCRIPTION
**Pull request by weirayao**
Change normalization from divided by the number of masked tokens to divided by total number of tokens. Justification is `dsigma` already accounts for the number of masked tokens. This is a hack to get something like per token loss. 

References:

[1] https://github.com/ML-GSAI/SMDM/blob/main/pretrain/train_mdm_rl.py#L281-L283
[2] https://github.com/ML-GSAI/LLaDA/blob/main/GUIDELINES.md#pre-training

### CODE REVIEW

**Review by shi-yu-wang — APPROVED** [review]
LGTM. I think it's essentially calculating the expection function (avergaing over number of sequences and sequence length) of the loss, as Eq. (3) in https://arxiv.org/pdf/2502.09992
